### PR TITLE
Fix incorrect amount of patients

### LINKED ISSUE #596: Incorrect number of Patients in Reference App
**Describe the bug**
There are 20 patients in the Patient's List view but on top it is showing 24 Patient(s) in textview.

**Component**
reference app

**To Reproduce**
1. Go to 'reference app'
2. See error

**Expected behavior**
It should have shown 20 Patient(s).

**Screenshots**
Not needed as such.

**Smartphone (please complete the following information):**
 - Device: xiaomi mi a3
 - Android version: 11
 - Build number: 11 RKQ1.200903.002 V12.0.9.0.RFQMIXM

**Additional context**
N/A

**Would you like to work on the issue?**
Yes

### PR DESCRIPTION
**Pull request by FikriMilano**
**IMPORTANT: All PRs must be linked to an issue (except for extremely trivial and straightforward changes).**

Fixes #596

**Description**
Change maximum patient list from 20 to real amount of patient, which is sourced from count() and also add the same filter as count(). So, the amount of patient displayed on the top TextView is the same as the patient list, depending how many patient match the filter.

**Alternative(s) considered**
N.A

**Type**
Choose one: Bug fix

**Screenshots (if applicable)**

**Checklist**
- [x] I have read and acknowledged the [Code of conduct](https://github.com/google/android-fhir/blob/master/docs/CODE_OF_CONDUCT.md)
- [x] I have read [How to Contribute](https://github.com/google/android-fhir/blob/master/docs/contributing.md)
- [x] I have read the [Developer's guide](https://github.com/google/android-fhir/wiki/Developer's-Guide)
- [x] I have signed the Google [Individual CLA](https://cla.developers.google.com/about/google-individual), or I am covered by my company's [Corporate CLA](https://cla.developers.google.com/about/google-corporate )
- [x] I have discussed my proposed solution with code owners in the linked issue(s) and we have agreed upon the general approach
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the style guide of this project
- [x] I have run `./gradlew check` and `./gradlew connectedCheck` to test my changes locally
- [x] I have built and run the reference app(s) to verify my change fixes the issue and/or does not break the reference app(s)

### CODE REVIEW

**Comment by Tarun-Bhardwaj** [comment]
@FikriMilano , is this PR ready for review?

**Comment by FikriMilano** [comment]
@Tarun-Bhardwaj , yes this PR is ready for review

File: reference/src/main/java/com/google/android/fhir/reference/PatientListViewModel.kt [hunk 1]
```diff
@@ -71,11 +71,15 @@ class PatientListViewModel(application: Application, private val fhirEngine: Fhi
             modifier = StringFilterModifier.CONTAINS
             value = nameQuery
           }
+        filter(Patient.ADDRESS_CITY) {
```
Comment (stevenckngaa) [inline reference/src/main/java/com/google/android/fhir/reference/PatientListViewModel.kt:74]: Using the patent count looks good. However, I am not very sure about the filter in count() and here. There isn't an explantation in https://github.com/google/android-fhir/pull/460/.

Anyway, could you please extract the city filter into a private function and call it  from count() and here.

**Review by stevenckngaa — COMMENTED** [review]
Thank you.

File: reference/src/main/java/com/google/android/fhir/reference/PatientListViewModel.kt (hunk 1 above)
Comment (Tarun-Bhardwaj) [reply reference/src/main/java/com/google/android/fhir/reference/PatientListViewModel.kt:74]: @FikriMilano , did you get a chance to work on the changes suggested by Steven?

Comment (FikriMilano) [reply reference/src/main/java/com/google/android/fhir/reference/PatientListViewModel.kt:74]: Sorry for the delay. I have been busy on other things lately, so I didn't have the chances to try this out yet. 

I should be able to continue working on this PR around this weekend.

**Comment by FikriMilano** [comment]
@stevenckngaa could you please review this again? thanks

File: reference/src/main/java/com/google/android/fhir/reference/PatientListViewModel.kt (hunk 1 above)
Comment (FikriMilano) [reply reference/src/main/java/com/google/android/fhir/reference/PatientListViewModel.kt]: Okay, done

> 
> 
> Using the patent count looks good. However, I am not very sure about the filter in count() and here. There isn't an explantation in #460.
> 
> Anyway, could you please extract the city filter into a private function and call it from count() and here.

**Review by stevenckngaa — APPROVED** [review]
Thank you!

**Comment by FikriMilano** [comment]
@stevenckngaa 
Is something holding this pr to get merged?

**Comment by stevenckngaa** [comment]
> @stevenckngaa
> Is something holding this pr to get merged?

Sorry, I was out of office. I approved this change 21 days ago. Feel free to submit.

**Comment by Tarun-Bhardwaj** [comment]
> > @stevenckngaa
> > Is something holding this pr to get merged?
> 
> Sorry, I was out of office. I approved this change 21 days ago. Feel free to submit.

@FikriMilano , when do you plan to merge these changes?

**Comment by FikriMilano** [comment]
> 
> 
> > > @stevenckngaa
> > > Is something holding this pr to get merged?
> > 
> > 
> > Sorry, I was out of office. I approved this change 21 days ago. Feel free to submit.
> 
> @FikriMilano , when do you plan to merge these changes?

@Tarun-Bhardwaj , I don't have write access so I can't merge PRs. 
@stevenckngaa could you please help help me with this?

**Comment by stevenckngaa** [comment]
> > > > @stevenckngaa
> > > > Is something holding this pr to get merged?
> > > 
> > > 
> > > Sorry, I was out of office. I approved this change 21 days ago. Feel free to submit.
> > 
> > 
> > @FikriMilano , when do you plan to merge these changes?
> 
> @Tarun-Bhardwaj , I don't have write access so I can't merge PRs.
> @stevenckngaa could you please help me with this?

Done. Thank you.

**Pull request merged**